devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectMojoBase.java

```diff
@@ -164,9 +168,36 @@ private String resolveValue(String expr) throws IOException {
         return expr;
     }
 
-    private List<Dependency> getManagedDependencies() throws IOException {
-        final DependencyManagement managed = project.getModel().getDependencyManagement();
-        return managed != null ? managed.getDependencies()
-                : Collections.emptyList();
+    private List<Dependency> getManagedDependencies(MavenArtifactResolver resolver) throws IOException {
+        List<Dependency> managedDependencies = new ArrayList<>();
+        Model model = project.getOriginalModel();
+        DependencyManagement managed = model.getDependencyManagement();
+        if (managed != null) {
+            managedDependencies.addAll(managed.getDependencies());
+        }
+        Parent parent;
+        while ((parent = model.getParent()) != null) {
+            try {
+                ArtifactDescriptorResult result = resolver.resolveDescriptor(new DefaultArtifact(
```

Comment:
I don't think you need the descriptor (effective POM). It looks like you need only to resolve the pom artifact with resolver.resolve(Artifact);

Reply:
Done, thanks!